src/utils.ts

```diff
@@ -94,6 +99,45 @@ class Utils {
   public static toBytes(hex: string): Uint8Array {
     return Uint8Array.from(Buffer.from(hex, "hex"));
   }
+
+  /**
+   * Convert the given transaction blob to a transaction hash.
+   *
+   * @param transactionBlobHex A hexadecimal encoded transaction blob.
+   * @returns A hex encoded hash if the input was valid, otherwise undefined.
+   */
+  public static transactionBlobToTransactionHash(
+    transactionBlobHex: string
+  ): string | undefined {
+    if (!isHex(transactionBlobHex)) {
+      return undefined;
+    }
+
+    const prefixedTransactionBlob = this.toBytes(
+      "54584E00" + transactionBlobHex
+    );
+    const hash = this.sha512Half(prefixedTransactionBlob);
+    return this.toHex(hash);
+  }
+
+  /**
+   * Compute the SHA512 half hash of the given bytes.
+   *
+   * @param input The input to hash.
+   * @returns The hash of the input.
+   */
+  private static sha512Half(bytes: Uint8Array): Uint8Array {
+    const sha512 = createHash("sha512");
+    const hashHex = sha512
+      .update(bytes)
+      .digest("hex")
+      .toUpperCase();
+    const hash = this.toBytes(hashHex);
+
+    /* eslint-disable @typescript-eslint/no-magic-numbers */
+    return hash.slice(0, hash.length / 2);
```

Comment:
NIT
I believe a sha512 hash always has a length of 64 so you could just write 32 instead of length / 2

Reply:
done